yelp_scrape.py

```diff
@@ -63,12 +64,40 @@ def collect_webpages(soup: BeautifulSoup, search_item: str) -> set:
     biz_lst: list = []
     for a in soup.find_all('a'):
         if 'href' in a.attrs:
-            if a.attrs['href'].split('?')[-1] == 'osq=' + search_item:
+            if a.attrs['href'].split('?')[-1] == 'osq=' + search_item and a.attrs['href'] not in biz_lst:
                 biz_lst.append(a.attrs['href'])
-    return set(biz_lst)
-
-
+        if len(biz_lst) >= 5:
+            break
+    return biz_lst
 
+def page_info_grab(pages: list) -> dict:
+    '''
+    iterate through the top search results obtained from "collect_webpages()" and pull relevant data to place into reccommndation info
+    :param list pages: list of urls of the top 5 search results from yelp
+    '''
+    reccommendation_info = {}
+    base_url = 'https://yelp.com'
+    for page in pages:
+        sleep(1)
+        page_attributes = {}
+        url = base_url + page
+        page_req = requests.get(url)
+        page_soup = BeautifulSoup(page_req.text, 'html.parser')
+        title = page_soup.h1.text
+        for label in page_soup.find_all('div'):   
+            if label.get('aria-label') is not None:
+                page_attributes['Rating'] = label['aria-label']
```

Comment:
So you're storing data in a dictionary with keys.

That's actually really common.

But now imagine someone sees a fully-documented version of your function that says `page_info_grab(pages: list[str]) -> dict[str, Any]`

How will they know what keys are valid, and what type will be returned when you provide a key?
There's not a good way to know what information they'll have access to when the function returns.

The answer to this problem is a `class`, in particular, I prefer `dataclass`es.

Here's the documentation: https://docs.python.org/3/library/dataclasses.html

Here's an example:
```
from dataclasses import dataclass

@dataclass(frozen=True)
class YelpInfo:
  rating: str
  website: str
  phone_number: str
  address: str
```

then you can create a new dataclass like this:
`info: YelpInfo = YelpInfo(rating=x, website=y, phone_number=z, address=w)`

Your function can now look like `def page_info_grab(...) -> YelpInfo:`
This makes it easy to know what's going on. You can just ctrl+click on the `YelpInfo`  part and it'll take them to the class definition where they can see what information exists
It's also common to add comments above each field describing the field in more detail (for example, does the phone number have the country code or not? maybe only for certain countries? how is it formatted? 1-234-567-8901 vs +01-(234)-567-8901, etc)

You do have to be a little careful though. When you mark a dataclass as `frozen=True` it means the data can't be changed after you make the class.

This has a number of advantages, it makes reasoning about your program much easier because you know once it has some data, it will always have that data. It just means that when constructing the class, you have to collect all the information together, _then_ call `YelpInfo(x, y, z, w)`